pytorch_lightning/overrides/base.py

```diff
@@ -54,15 +54,15 @@ def forward(self, *inputs, **kwargs):
             # ddp_plugin ``post_training_step`` hook
             if not self.module.automatic_optimization:
                 self.module.trainer.model.require_backward_grad_sync = False
-            warn_if_output_is_none(output, "training_step")
+
+            if self.module.automatic_optimization:
+                warn_if_output_is_none(output, "training_step")
```

Comment:
I wonder why do we have this warning here. Isn't this enough? https://github.com/PyTorchLightning/pytorch-lightning/blob/0456b4598f5f7eaebf626bca45d563562a15887b/pytorch_lightning/trainer/training_loop.py#L756-L757

Reply:
could you include this in the test below? automatic -> warning, manual -> no warning
thanks

Reply:
yes doing that. but my question here is if we already have a similar warning inside training_loop, then why do we need it in overrides/base.py?

Reply:
Most probably because this was added and nobody remembered the other one :D

Although IMO it fits better in the training_loop than here

Reply:
yeah, since this one is somewhat specific to distributed training, just wanted to confirm whether to remove it or not. @awaelchli

Reply:
hehe it just looks like I added it but it was a refactor :) It was there originally. 
This is a leftover artifact of how we called the training_step before in distributed and DP setting but now it should not be needed IF we already do it in the training loop. 

For DP I will check tomorrow how it behaves, not sure there because of the gather... Have to test.

Reply:
hi @awaelchli, did you check this for DP?

Reply:
@rohitgr7  sorry for the delay. 
I tested and I get two different warnings for DP:

```
/home/adrian/.conda/envs/lightning/lib/python3.7/site-packages/pytorch_lightning/utilities/distributed.py:50: UserWarning: Your training_step returned None. Make sure that was your intention!
  warnings.warn(*args, **kwargs)


/home/adrian/.conda/envs/lightning/lib/python3.7/site-packages/pytorch_lightning/utilities/distributed.py:50: UserWarning: training_step returned None if it was on purpose, ignore this warning...
  warnings.warn(*args, **kwargs)

```
It should be fine to remove the one in the module completely (we can still have tests for the warning in the training loop).

Reply:
cool! thanks.. will update.